File: APISolidarityManager/APISolidarityManager/Entities/NeedRule.cs
namespace APISolidarityManager.Models
{
    public class NeedRule
    {
        public Guid Id { get; set; }
        public Guid AgeRangeId { get; set; }
        public Guid NeedGroupId { get; set; }
        public decimal Value { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }

        public AgeRange AgeRange { get; set; } = null!;
        public NeedGroup NeedGroup { get; set; } = null!;
    }
}

File: donation-system-web/src/types/needRule.ts
export interface NeedRuleResponse {
  id: string;
  ageRangeId: string;
  ageRangeName: string;
  needGroupId: string;
  needGroupName: string;
  value: number;
  createdAt: string;
  updatedAt?: string;
}

export interface CreateNeedRuleRequest {
  ageRangeId: string;
  needGroupId: string;
  value: number;
}

export interface UpdateNeedRuleRequest {
  ageRangeId: string;
  needGroupId: string;
  value: number;
}
